src/web/src/app/space/detail/component.html

```diff
@@ -0,0 +1,113 @@
+<!-- end of listing invitations -->
+
+<hr class="width-and-clear">
+<div class="space-links-heading">
+	<h2 i18n="@@SPACES.DETAIL.REFS.HEADING">Ссылки-приглашения</h2>
+	<button mat-icon-button (click)="onLinkAdd()"><mat-icon>add</mat-icon></button>
+</div>
+<div class="width-and-clear">
+	<!-- form for creating link -->
+	<div class="row">
+		<form [formGroup]="linkForm" [hidden]="isLinkFormHidden" (ngSubmit)="onSubmitLink(linkForm)">
+			<div class="left-float left-padding">
+				<mat-form-field appearance="fill">
+					<input matInput type="text" formControlName="name" required />
+					<mat-hint i18n="">Имя ссылки</mat-hint>
+				</mat-form-field>
+			</div>
+			<div class="left-float left-padding">
+				<mat-form-field appearance="fill">
+					<input matInput type="date" formControlName="expiredAt" required />
+					<mat-hint i18n="">Срок действия</mat-hint>
+				</mat-form-field>
+			</div>
+			<div class="left-float left-padding"><button mat-icon-button type="submit" i18n=""><mat-icon>done</mat-icon></button></div>
+			<div class="left-float left-padding"><button mat-icon-button (click)="onLinkFormCloseBtn()"><mat-icon>close</mat-icon></button></div>
+		</form>
+	</div> <!-- end form for creating link -->
+	
+	<!-- listing links -->
+	<div *ngFor="let link of links" class="row">
+		<div class="left-float">{{link.name}}</div>
+		<div class="left-float left-padding">{{link.expiredAt | date}}</div>
+		<div class="left-float left-padding"><button mat-raised-button i18n="@@SPACES.DETAIL.REFS.COPYBTN" (click)="onLinkCopyBtn(link)">Копировать</button></div>
+		<div class="left-float left-padding"><button mat-icon-button (click)="onLinkDelBtn(link)"><mat-icon>close</mat-icon></button></div>
+	</div>
+</div>
+<mat-paginator #linksPaginator class="width-and-clear" [length]="linksTotal" [pageSize]="linksPageSize" [pageSizeOptions]="[7, 15, 30, 100]" (page)="getLinks($event.pageSize, $event.pageIndex); savePageSizeToLocalStorage('linksPaginatorSpaceDetail', $event);"></mat-paginator> 
+<!-- end of listing links -->
+
+<!-- listing users -->
+<hr class="width-and-clear">
+<h2 i18n="@@SPACES.DETAIL.USERS.HEADING">Пользователи</h2>
+<div class="width-and-clear">
+	<div *ngFor="let user of users" class="row">
+		<div class="left-float">{{getUserNameById(user.userId)}}</div>
+		<div class="left-float left-padding">{{getUsernameById(user.userId)}}</div>
+		<div class="left-float left-padding">
+			<mat-form-field>
+				<mat-select matNativeControl [(ngModel)]="user.role">
+					<mat-option value="admin" i18n="@@SPACES.DETAIL.ADMINROLE">Администратор</mat-option>
+					<mat-option value="user" i18n="@@SPACES.DETAIL.USERROLE">Пользователь</mat-option>
+					<mat-option value="guest" i18n="@@SPACES.DETAIL.GUESTROLE">Гость</mat-option>
+				</mat-select>
+			</mat-form-field>
+		</div>
+		<div class="left-float left-padding">{{getUserEmailById(user.userId)}}</div>
```

Comment:
А для чего тут matNativeControl?

Давай пожалуйста сделаем полную инфу о пользователе во всплывающей подсказке.


```suggestion
		<div class="left-float"
			[matTooltip]="user.login + ' ' + user.email">{{user.displayName}}</div>
		<div class="left-float left-padding">
			<mat-form-field>
				<mat-select [(ngModel)]="user.role">
					<mat-option value="admin" i18n="@@SPACES.DETAIL.ADMINROLE">Администратор</mat-option>
					<mat-option value="user" i18n="@@SPACES.DETAIL.USERROLE">Пользователь</mat-option>
					<mat-option value="guest" i18n="@@SPACES.DETAIL.GUESTROLE">Гость</mat-option>
				</mat-select>
			</mat-form-field>
		</div>
```

Reply:
matNativeControl - это рудимент от старой реализации селекта ("некрасивого"). Удалил.

matTooltip реализовал - не очень красиво, но работает.